editor/components/block-toolbar/style.scss

```diff
@@ -13,7 +13,11 @@
 
 	.components-toolbar {
 		border: none;
-		border-left: 1px solid $light-gray-500;
+	}
+
+	// Only show the left border if the Switcher is present
+	.editor-block-switcher + div > .components-toolbar {
```

Comment:
This seems very prone to future breakage, particularly on the `+ div` sibling part, since the `div` it refers to exists only for adding a left border in a set of toolbar controls. This has no reason for needing to be a `div` specifically, and could otherwise be refactored to a solution which no longer requires the `div` wrapper at all.

Instead, I'd like to see one of:

- Adding a `border-right` to the `Toolbar` of the `EditorBlockSwitcher`
- See what we can do about removing [this "seemingly redundant wrapper" `div`](https://github.com/WordPress/gutenberg/blob/6728906116c12b5865e66b32b634d2b300136334/components/dropdown/index.js#L75-L79) and change selector to apply `border-left` on `.editor-block-toolbar .components-toolbar:not( :first-child )`
   - Alas, we can't use [`:first-of-type`](https://developer.mozilla.org/en-US/docs/Web/CSS/:first-of-type) for anything except an element tag ([reference](https://stackoverflow.com/a/6447072/995445)). Well, maybe [someday](https://stackoverflow.com/a/19474149/995445)).
- Add `hasLeftDivider` as a prop to control / override the existing behavior of `Toolbar` in applying the `.has-left-divider` modifier class

Reply:
The first idea was a solid bit of advice. Did that:

![border fix](https://user-images.githubusercontent.com/1204802/40828899-3178e5b2-6582-11e8-80bf-bbc0f2c114b1.gif)